homeassistant/components/light/device_action.py

```diff
@@ -119,15 +144,14 @@
     elif entry:
         supported_features = entry.supported_features
 
-    if not supported_features & SUPPORT_BRIGHTNESS:
-        return {}
+    extra_fields = vol.Schema({})
 
-    return {
-        "extra_fields": vol.Schema(
-            {
-                vol.Optional(ATTR_BRIGHTNESS_PCT): vol.All(
-                    vol.Coerce(int), vol.Range(min=0, max=100)
-                )
-            }
+    if supported_features & SUPPORT_BRIGHTNESS:
+        extra_fields = extra_fields.extend(
+            {vol.Optional(ATTR_BRIGHTNESS_PCT): VALID_BRIGHTNESS_PCT}
         )
-    }
+
+    if supported_features & SUPPORT_FLASH:
+        extra_fields = extra_fields.extend({vol.Optional(ATTR_FLASH): VALID_FLASH})
+
+    return {"extra_fields": extra_fields}
```

Comment:
Instead of extending a `vol.Schema` object, just keep the initial `extra_fields` as a dictionary. Then change this line to:
```suggestion
    return {"extra_fields": vol.Schema(extra_fields)} if extra_fields else {}
```

Reply:
Changed